packages/tanstack-start/src/server/middlewareHandler.ts

```diff
@@ -0,0 +1,44 @@
+import type { AnyRouter } from '@tanstack/react-router';
+import type { EventHandler } from 'vinxi/http';
+
+import { authenticateRequest } from './authenticateRequest';
+import { loadOptions } from './loadOptions';
+import { getResponseClerkState } from './utils';
+
+export type RequestHandler<TRouter extends AnyRouter> = (ctx: {
+  request: Request;
+  router: TRouter;
+  responseHeaders: Headers;
+}) => Promise<Response>;
+export type CustomizeRequestHandler<TRouter extends AnyRouter> = (cb: RequestHandler<TRouter>) => EventHandler;
+
+export function clerkMiddlewareHandler<TRouter extends AnyRouter>(eventHandler: CustomizeRequestHandler<TRouter>) {
```

Comment:
Could someone pass options to this one? For example, is it possible to set a PK/SK pair during runtime instead of simply relying on env vars?

Reply:
I didn't want to introduce anything for now as we don't know if this is the final solution.

Reply:
Good thinking :) Im raising this because it could result in an public API change so lets see an example of passing options to the helper first. Im thinking that this:

```ts
const handler = createRequestHandler({
  createRouter,
  getRouterManifest,
});

export default clerkMiddlewareHandler(handler, { secretKey: '....' })(defaultStreamHandler);
```

might turn out to be a bit too complicated for us. If we want to accept options, we could maybe turn the helper into a factory function, ex:

```
const handler = createRequestHandler({
  createRouter,
  getRouterManifest,
});

const clerkHandler = createClerkHandler(handler, { secretKey: '...' })

export default clerkHandler(defaultStreamHandler);
```
or similar

Reply:
Yeah it looks good, I will implement this today and see if we can go with a better solution